### PR TITLE
[XHamster] fix: extract uploader_id

### PR DESCRIPTION
**Pull request by octotherp**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

### CODE REVIEW

**Comment by pukkandan** [comment]
Why is uploader same as uploader_id? From the tests, it doesn't look like it is the ID that is being extracted as `uploader`

**Comment by octotherp** [comment]
There is no uploader_id in tests and no uploader_id in previous version.
There is uploader_id in site pages but it is pretty much a meaningless internal int id, API works through `lowercase(uploader)`. 
This patch is mostly a workaround so `--output '%(extractor)s/%(uploader_id)s/%(title)s-%(id)s.%(ext)s'` would work uniformely on all extractors (that I use) as there no way to configure yt-dlp per extractor and no way to do `uploader_id?uploader_id:uploader` in output formatter and it is better than empty uploader_id.

File: yt_dlp/extractor/xhamster.py [hunk 1]
```diff
@@ -245,14 +245,16 @@ def get_height(s):
             else:
                 categories = None
 
+            uploader = try_get(video, lambda x: x['author']['name'], compat_str)
```
Comment (pukkandan) [inline yt_dlp/extractor/xhamster.py:248]:
```suggestion
            uploader_url = url_or_none(try_get(video, lambda x: x['author']['pageURL']))
```

File: yt_dlp/extractor/xhamster.py [hunk 2]
```diff
@@ -245,14 +245,16 @@ def get_height(s):
...
             return {
                 'id': video_id,
                 'display_id': display_id,
                 'title': title,
                 'description': video.get('description'),
                 'timestamp': int_or_none(video.get('created')),
-                'uploader': try_get(
-                    video, lambda x: x['author']['name'], compat_str),
+                'uploader': uploader,
+                'uploader_id': uploader,
```
Comment (pukkandan) [inline yt_dlp/extractor/xhamster.py:257]:
```suggestion
                'uploader': try_get(
                    video, lambda x: x['author']['name'], compat_str),
                'uploader_url': uploader_url,
                'uploader_id': uploader_url.split('/')[-1] if uploader_url else None,
```

File: yt_dlp/extractor/xhamster.py [hunk 3]
```diff
@@ -352,6 +354,7 @@ def get_height(s):
             'description': description,
             'upload_date': upload_date,
             'uploader': uploader,
+            'uploader_id': uploader,
```
Comment (pukkandan) [inline yt_dlp/extractor/xhamster.py:357]:
```suggestion
            'uploader_id': uploader.lower() if uploader else None,
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Pull request merged**